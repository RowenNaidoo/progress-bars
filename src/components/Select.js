import React from 'react'
import PropTypes from 'prop-types';
import styled from 'styled-components'

const Select = ({ options, selectChanged }) => (
  <StyledSelect
    onChange={(e) => { selectChanged(e.target.value) }}>
    {options.map((opt, index) => (
      <option key={`Option_${index}`} value={index}>{`# progress ${index + 1}`}</option>
    ))}
  </StyledSelect>
)

const StyledSelect = styled.select`
  width: 100%;
  height: 40px;
`

Select.propTypes = {
  options: PropTypes.arrayOf(PropTypes.number),
  selectChanged: PropTypes.func
}

export default Select