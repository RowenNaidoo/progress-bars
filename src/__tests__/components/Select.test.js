import React from 'react'
import { shallow } from 'enzyme'
import Select from '../../components/Select'


describe('#Select', () => {

  test('renders learn react link', () => {
    const component = shallow(<Select options={[39, 37]} />)
    expect(component).toMatchSnapshot()
  })

})