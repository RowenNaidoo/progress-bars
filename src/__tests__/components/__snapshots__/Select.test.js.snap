// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Select renders learn react link 1`] = `ShallowWrapper {}`;
